week2/homework/src/add.js

```diff
@@ -0,0 +1,13 @@
+function add(data, args) {
+  return new Promise(resolve => {
+    let keyOfNewItem = Object.keys(data).length + 1;
```

Comment:
This code looks like it  calculates the next available index i.e. key. But I do not understand why do you need the while loop then? Is one of them not working or is there another reason?

Reply:
Hello @gajduk !! thanks for the feedback and any i really appreciate your time and effort.

 it calculates the available index based on the length of the object. suppose that i have an object with 3 properties. 
( { "1" : "goto market" , "2" : "visit mom", "3" : "see the dentist"} ). so the next property will be length of my object (3) +1 =4. But suppose that i removed 2 from the list. then my object is:
( { "1" : "goto market" , "3" : "see the dentist"} ) So when i add new item, the key will be length of my object(2) +1 =3. But 3 is already in the list("3" : "see the dentist"). So i check if the new key is in the list with: .hasOwnProperty(key). if the newly created key exists in the list, it is incremented in the while loop until it doesnt conflict with any of the key in the object.

By the way my code works fine in my computer and does not return any error. sorry for the inconvenience. I checked the displayList method and I think the error is caused by not handling rejected state of the promise. so i added try-catch block. I am not able to test it, becuz i dont get any error. But anyway I will make a new pull request. I hope it works fine.
thank you very much again
Kind regards